sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -335,6 +358,19 @@ trait Row extends Serializable {
   def getAs[T](fieldName: String): T = getAs[T](fieldIndex(fieldName))
 
   /**
+   * Returns the value of a given fieldName as a scala.util.Try object.
+   */
+  def attempt[T](fieldName: String): Try[T] = Try(getAs[T](fieldIndex(fieldName)))
```

Comment:
Yeah, sorry if I wasn't being clear earlier.  There certainly is still room for improvement here if you are still looking to contribute.  Some ideas:
 - That error message doesn't tell you which column the problem happens in.
 - Its using an old reflection based codepath instead of encoders (which would probably be much faster).
 - Here, it might be nice to have the option to get `nulls` instead of exceptions, though this would have to be optional.